pkg/server/status.go

```diff
@@ -1515,39 +1567,23 @@ func (s *statusServer) fetchProfileFromAllNodes(
 	errorFn := func(nodeID roachpb.NodeID, err error) {
 		response.profDataByNodeID[nodeID] = &profData{err: err}
 	}
-	if err := s.iterateNodes(ctx, "cluster-wide CPU profile", dialFn, nodeFn, responseFn, errorFn); err != nil {
+	if err := s.iterateNodes(ctx, opName, dialFn, nodeFn, responseFn, errorFn); err != nil {
 		return nil, serverError(ctx, err)
 	}
-
-	profs := make([]*profile.Profile, 0, len(response.profDataByNodeID))
-	for nodeID, pd := range response.profDataByNodeID {
-		if len(pd.data) == 0 && pd.err == nil {
-			log.Warningf(ctx, "no profile collected for node %d", nodeID)
-			continue // skipped node
-		}
-
-		if pd.err != nil {
-			log.Warningf(ctx, "failed to collect profile for node %d: %v", nodeID, pd.err)
-			continue
-		}
-
-		p, err := profile.ParseData(pd.data)
-		if err != nil {
-			return nil, err
-		}
-		p.Comments = append(p.Comments, fmt.Sprintf("n%d", nodeID))
-		profs = append(profs, p)
+	var data []byte
+	switch req.Type {
+	case serverpb.ProfileRequest_CPU:
+		data, err = s.processCPUProfilesFromAllNodes(ctx, response)
+	case serverpb.ProfileRequest_GOROUTINE:
+		data, err = s.processGoroutineProfilesFromAllNodes(ctx, response)
+	default:
+		return nil, errors.Newf("cluster-wide collection of %s is unsupported", req.Type.String())
```

Comment:
```suggestion
		return nil, errors.Newf("cluster-wide collection of %s is unsupported", req.Type)
```

I fully expect you'll come back to me saying that the `.String` is necessary for some reason, but it shouldn't be. Perhaps `String` needs to be implemented on the pointer instead of value receiver or vice versa, either way we shouldn't have to call it explicitly here.

Reply:
I don't think I had a good reason to add .String suffixes. Dropped them and the test that checks for this type of error during request validation still passes.